firebase_admin/_auth_utils.py

```diff
@@ -275,6 +277,131 @@ def validate_provider_ids(provider_ids, required=False):
+    # validate multiFactorConfig
+    if not isinstance(mfa_config, dict):
+        raise ValueError(
+            'multiFactorConfig should be of valid type MultiFactorConfig')
+
+    #validate MFA Config keys
+    valid_mfa_config_keys = set(['state', 'factorIds', 'providerConfigs'])
+    mfa_config_keys = set(mfa_config.keys())
+    for key in mfa_config_keys:
+        if key not in valid_mfa_config_keys:
+            raise ValueError('{0} is not a valid MultiFactorConfig paramter'.format(key))
+
+    # validate state in MFA config
+    if 'state' not in mfa_config:
+        raise ValueError('multiFactorConfig.state should be defined')
+    state = mfa_config['state']
+    if not isinstance(state, str) or state not in VALID_STATES:
+        raise ValueError(
+            'multiFactorConfig.state must be either "ENABLED" or "DISABLED"')
+    mfa_config_payload['state'] = state
+
+    # validate factorIds if defined
+    if 'factorIds' in mfa_config:
+        factor_ids = mfa_config['factorIds']
+        if not isinstance(factor_ids, list) or not factor_ids:
+            raise ValueError(
+                'multiFactorConfig.factorIds must be a defined list of AuthFactor type strings')
+        for factor_id in factor_ids:
+            if not isinstance(factor_id, str) or factor_id not in VALID_AUTH_FACTOR_TYPES:
+                raise ValueError(
+                    'factorId must be a valid AuthFactor type string')
+        mfa_config_payload['enabledProviders'] = factor_ids
+
+    # validate providerConfigs if defined
+    if 'providerConfigs' in mfa_config:
+        provider_configs = mfa_config['providerConfigs']
+        provider_configs_payload = []
+        if not isinstance(provider_configs, list) or not provider_configs:
+            raise ValueError(
+                'multiFactorConfig.providerConfigs must be a valid list of providerConfig types')
+        for provider_config in provider_configs:
+            provider_config_payload = {}
+            if not isinstance(provider_config, dict) or not provider_config:
+                raise ValueError(
+                    'multiFactorConfigs.providerConfigs must be a valid array of '
+                    'type providerConfig')
+
+            #validate Provider Config keys
+            valid_provider_config_keys = set(['state', 'totpProviderConfig'])
+            provider_config_keys = set(provider_config.keys())
+            for key in provider_config_keys:
+                if key not in valid_provider_config_keys:
+                    raise ValueError('{0} is not a valid ProviderConfig paramter'.format(key))
+
+            if 'state' not in provider_config:
+                raise ValueError('providerConfig.state should be defined')
+            state = provider_config['state']
+            if not isinstance(state, str) or state not in VALID_STATES:
+                raise ValueError(
+                    'providerConfig.state must be either "ENABLED" or "DISABLED"')
```

Comment:
nit: "not in " + str(VALID_STATES)

Reply:
I think consistency across SDKs isn't a huge issue. By using the variable, you don't need to edit the error message everytime we add to it. Probably more applicable to VALID_AUTH_FACTOR_TYPES than VALID_STATES.